vortex-array/src/patches.rs

```diff
@@ -636,19 +639,45 @@ mod test {
     #[test]
     fn search_right() {
         let patches = Patches::new(
-            2,
+            6,
             0,
-            buffer![0u64].into_array(),
-            PrimitiveArray::new(buffer![0u8], Validity::AllValid).into_array(),
+            buffer![0u8, 1, 4, 5].into_array(),
+            buffer![-128i8, -98, 8, 50].into_array(),
         );
 
         assert_eq!(
-            patches.search_sorted(0, SearchSortedSide::Right).unwrap(),
-            SearchResult::Found(1)
+            patches.search_sorted(-98, SearchSortedSide::Right).unwrap(),
+            SearchResult::Found(2)
```

Comment:
why is 2 the correct value and not 1? The doc comment for SearchResult::Found is `Result for a found element was found at the given index in the sorted array`

Reply:
If you’re searching from Right then you want the index one after the equal index. We replicate numpy semantics https://numpy.org/doc/2.0/reference/generated/numpy.searchsorted.html

Reply:
ok, we should update the doc comment